### ISSUE TITLE
panel crashes on startup under Arch

### ISSUE THREAD

**Issue by danboid** [solus-project/budgie-desktop#302]
I currently cannot login to Budgie under Arch. It was working last night. I added a few shortcuts and a workspace switcher to the panel and after logging out or rebooting I can't log into budgie any more.

Here is a `journalctl -ae` dump:

https://gist.github.com/danboid/7b71a7eb93a611326763

**Comment by ikeydoherty** [comment]
```
Jan 03 11:31:55 arch76 gnome-session[20228]: (budgie-panel:20290): Gtk-ERROR **: GtkBox child WorkspacesApplet minimum width: -1 < 0 for height 30
Jan 03 11:31:55 arch76 gnome-session-binary[20228]: WARNING: Application 'budgie-panel.desktop' killed by signal 5
```

Interesting bug. For now, reset dconf for Budgie per the README.md.I'm guessing you set workspaces to 0 in gnome-tweak-tool :)

**Comment by danboid** [comment]
No, I didn't set workspaces to 0 with gnome tweak.

If I reset budgies dconf I can log in. If I then add a workspace switcher and leave it at the default setting of 4 workspaces then log out I cannot log in and I have to reset dconf again.

**Comment by ikeydoherty** [comment]
Can you try with that fix applied please ^ ?

**Comment by danboid** [comment]
That seems to have fixed the prob with the workspace switcher but I'm getting another panel crash now. It's prob very closely related so I thought I'd keep it in this same issue.

If I pin an app to the panel then log out, the panel crashes on my next login attempt. Strangely enough though I can log in successfully on the 2nd attempt after pinning a new app.

https://gist.github.com/danboid/b2c21666dfbb8f596847

**Comment by danboid** [comment]
UPDATE

I was wrong with my description of my current panel crash. It seems that as long as I have at least one app pinned to the panel, the budgie panel crashes on the first login attempt with the errors from the previous post. Login  works on the second attempt but it does not depend on the using pinning additional apps to cause the crash on the first attempt.

**Comment by juvevood** [comment]
panel crashed in 10.2.2,(it works with v10.2.1), and i have 4 workspace.

**Comment by geeks-dev** [comment]
I'm having this same issue.
v10.2.1 In no problem.
I also workspace was set to 4 .

**Comment by danboid** [comment]
I've just updated to latest git. I can see some un-derping has happened as the panel crash happens less often now as I just did a whole bunch of reboots, shutdowns/startups and logging in and out and it only crashed once out of 10 or so of dem restarty tings.

**Comment by danboid** [comment]
With the previous build I was running from a few days ago login would fail with a crash pretty much every other attempt.

**Comment by geeks-dev** [comment]
It was a simple problem .
I was running `/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1 &` in .xprofile , but it was resolved by commenting out this .

I should have been aware earlier that the following warning was out to `journalctl` .

```
gnome-session-binary[1052]: WARNING: App 'budgie-polkit.desktop' exited with code 1
gnome-session-binary[1052]: WARNING: App 'budgie-polkit.desktop' respawning too quickly
gnome-session-binary[1052]: Unrecoverable failure in required component budgie-polkit.desktop                                                                                                 
gnome-session[1052]: gnome-session-binary[1052]: WARNING: App 'budgie-polkit.desktop' respawning too quickly
```

**Comment by danboid** [comment]
I don't have a ~/.Xprofile so I don't think that's what's causing my crash.

**Comment by ikeydoherty** [comment]
Please confirm if this is still an issue, given Budgie is in Arch I can't see how it is, so let me know by Friday.

**Comment by danboid** [comment]
Just built the latest budgie-desktop-git under Arch and I see no crashing panels Ikey!

:D

**Comment by ikeydoherty** [comment]
Great, thanks!!

**Comment by ikeydoherty** [comment]
Can someone report this upstream to Arch maintainers to stop disabling the workspaces applet in the community repo then?

**Comment by danboid** [comment]
I've just added a comment https://aur.archlinux.org/packages/budgie-desktop-git/

**Comment by danboid** [comment]
I probably closed this too soon. I forgot that I had to customise the panel before problems occurred.

On first boot, budgie (under Arch) was fine and even showed my Cinnamon desktop icons. Great! Then I changed a few things like added a workspace switcher to the panel and changed its position to be more to the left of the panel. On my next budgie login, the panel failed to load and I had no desktop icons at all so I rebooted. On the next attempt, budgie loaded and I could see the panel again but it had forgot the positioning of the workspace switcher (it was back in the centre) and my desktop shortcuts were missing. I logged out and back into cinnamon and my desktop shortcuts returned.

Re-opening but I don't mind if you want to close it and make a new ticket as it may not be the same issue?

**Comment by fossfreedom** [comment]
that git pkgbuild looks like its compiling directly from git rather thank ikey's strong recommendation to use make dist.

is there a stable AUR which has been built using Ikey's recommended tarball ?

**Comment by ikeydoherty** [comment]
@fossfreedom the git packages shouldn't be building from `master`, that's my main development branch. They should be using specific commits, testing, and then bumping.

@danboid I'm almost convinced this is an Arch specific issue now, as Ubuntu (via Budgie Remix) and Solus do not have this issue. Think its time to raise this issue upstream in Arch tbqh. I also need some stack traces if this is going to be in any way useful for me.

Use `coredumpctl gdb` to get me a trace of the last segfautl please

**Comment by City-busz** [comment]
This is the console output:
```
(budgie-panel:2148): Gtk-WARNING **: Allocating size to GtkBox 0x24a5ba0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?

(budgie-panel:2148): Gtk-ERROR **: GtkBox child WorkspacesApplet minimum width: -1 < 0 for height 33
Trace/breakpoint trap (core dumped)
```

And here is the trace:
[trace.txt](https://github.com/solus-project/budgie-desktop/files/380579/trace.txt)

**Comment by ikeydoherty** [comment]
Vala showing itself to be fully feckin useless, even the strace doesn't help, alright thanks @City-busz

**Comment by danboid** [comment]
I rebuilt budgie under a fully updated Arch today, customised a few things, tried logging in and out a few times, actually using it etc and I've not had any panel crashes yet. I'd give it a week before I'd be willing to close this after my last hasty closing incident.

I did however experience an issue where my desktop icons doubled up so that every shortcut appeared twice on the desktop. If it happens again and I work out what causes it I'll open a new ticket.

**Comment by ikeydoherty** [comment]
Please don't, Budgie isn't responsible for desktop icons. You probably have Nemo and Cinnamon installed in parallel or something.

**Comment by danboid** [comment]
Ah yes! The second set of icons would've appeared when I loaded nemo, which I definitely prefer over Files.

I'll have to create a shortcut to start Nemo in non desktop-grabby mode.

**Comment by danboid** [comment]
That's made me think - seeing as I prefer Nemo I'm prob better off disabling desktop icons in Budgie then getting Nemo to auto-start with Budgie in desktop mode. That should work, right?

**Comment by ikeydoherty** [comment]
Sounds like it yeah

**Comment by danboid** [comment]
I've not seen any panel crashes under Arch since my budgie-git rebuild last week so I'm happy to close this now.

Are you still having issues City-busz? I'll leave it to Ikey to decide if he wants to close this and create a new ticket for City-busz if they're still having probs.

**Comment by City-busz** [comment]
@danboid It was I who fixed the problem. ;) Feel free to close this task.

**Comment by danboid** [comment]
City-busz: Thanks! Closing.

**Comment by ikeydoherty** [comment]
Ya forgot to update this issue sorry, PR by @City-busz after fixed things on Arch.

**Comment by ikeydoherty** [comment]
So just as resolution for this issue.. This suddenly appeared in Solus today, just like magic.
It turns out the specific *cause* is that GTK 3.20 breaks it. Basically. So we've included this patch
now in the Solus build of `budgie-desktop`: https://git.solus-project.com/packages/budgie-desktop/commit/?h=budgie-desktop-10.2.6-91

> [22:12:23] <Guest69994> with workspaces only 1 out of 7 logins successfull, without not a single one failed
